### PR TITLE
[orc8r] DB Alerts

### LINKED ISSUE #6970: Add orc8r alerts in cloudwatch
- Add alerts for monitoring service health
- Add alerts for RDS failures

### PR DESCRIPTION
**Pull request by arunuke**
Signed-off-by: Arun Thulasi <arunt@fb.com>

## Summary

An SNS queue is set up for Orc8r DB to send notifications in the event of a failure. DB is also configured to take automatic daily backups.

## Test Plan

Deploy Orc8r with DB configured to send notifications to SNS with an email endpoint. Stop database and check if notifications are received. Restart database from a backed up instance. Startup database from a previously stored snapshot/backup to measure time.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: orc8r/cloud/deploy/terraform/orc8r-aws/db.tf [hunk 1]
```diff
@@ -26,8 +26,29 @@ resource "aws_db_instance" "default" {
...
+  backup_retention_period = var.orc8r_db_backup_retention
+  backup_window           = var.orc8r_db_backup_window
+
   skip_final_snapshot = true
   # we only need this as a placeholder value for `terraform destroy` to work,
   # this won't actually create a final snapshot on destroy
   final_snapshot_identifier = "foo"
 }
+
+resource "aws_sns_topic" "sns_orc8r_topic" {
```
Comment (karthiksubraveti) [inline orc8r/cloud/deploy/terraform/orc8r-aws/db.tf:38]: Can you add a boolean(aws_db_event_alerts) which is optional and turned off by default, which can gate these other variables. These lack of this mandatory variable(orc8r_sns_email) may cause errors in existing upgrades and will lead to unnecessary confusion.

File: orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf [hunk 2]
```diff
@@ -34,6 +35,7 @@ module "orc8r" {
   elasticsearch_ebs_enabled     = true
   elasticsearch_ebs_volume_size = 32
   elasticsearch_ebs_volume_type = "gp2"
+
```
Comment (karthiksubraveti) [inline orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf]: extra whiteline ?

**Review by karthiksubraveti — CHANGES_REQUESTED** [review]

**Review by karthiksubraveti — COMMENTED** [review]
Can you add the configuration vars to -> https://github.com/magma/magma/blob/master/orc8r/cloud/deploy/orc8r_deployer/docker/root/data/vars.yml and mark it required or optional accordingly ?

File: orc8r/cloud/deploy/terraform/orc8r-aws/db.tf (hunk 1 above)
Comment (arunuke) [reply orc8r/cloud/deploy/terraform/orc8r-aws/db.tf:38]: Can I set the boolean gate just for the mandatory variable (orc8r_sns_email)? The SNS and the event subscription would exist based on default values enabled in the configurations, but no notification would be sent unless the customer desires so. 

I would like to keep the backup option enabled to ensure we are protected against EC2 instance failure coupled with an irrecoverable EBS volume. WDYT?

Comment (arunuke) [reply orc8r/cloud/deploy/terraform/orc8r-aws/db.tf:38]: Added new flag enable_aws_db_notifications with 'false' as default.

**Comment by arunuke** [comment]
> Can you add the configuration vars to -> https://github.com/magma/magma/blob/master/orc8r/cloud/deploy/orc8r_deployer/docker/root/data/vars.yml and mark it required or optional accordingly ?

Added all new variables to vars.yml marked as "false" for required since they are all either specified as default or optional

File: orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf (hunk 2 above)
Comment (arunuke) [reply orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf]: Fixed.

File: orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf [hunk 3]
```diff
@@ -20,6 +20,7 @@ module "orc8r" {
   orc8r_db_password           = "mypassword" # must be at least 8 characters
   secretsmanager_orc8r_secret = "orc8r-secrets"
   orc8r_domain_name           = "orc8r.example.com"
+  orc8r_sns_email             = "admin@example.com"
```
Comment (karthiksubraveti) [inline orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf:23]: minor suggestion - add `enable_aws_db_notifications` also to this example.

**Review by karthiksubraveti — APPROVED** [review]

File: orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf (hunk 3 above)
Comment (arunuke) [reply orc8r/cloud/deploy/terraform/orc8r-helm-aws/examples/basic/main.tf:23]: Added enable_aws_db_notifications to main.tf

**Review by karthiksubraveti — APPROVED** [review]

File: orc8r/cloud/deploy/orc8r_deployer/docker/root/data/vars.yml [hunk 4]
```diff
@@ -74,6 +74,42 @@ infra:
...
+    Required: false
+    ConfigApps:
+      - tf
+  orc8r_db_event_subscription:
+    Description: Name for orc8r db event subscriber
+    Type: string
+    Required: false
+    ConfigApps:
+      - tf
+  orc8r_sns_mail:
```
Comment (quentinDERORY) [inline orc8r/cloud/deploy/orc8r_deployer/docker/root/data/vars.yml]: typo this should be orc8r_sns_email

Comment (arunuke) [reply orc8r/cloud/deploy/orc8r_deployer/docker/root/data/vars.yml]: Thanks @quentinDERORY . Fixed.

**Pull request merged**